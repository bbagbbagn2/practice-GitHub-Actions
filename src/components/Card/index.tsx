// src/components/Card/index.tsx

export default function Card({
  title,
  description,
  imageUrl,
}: {
  title: string;
  description: string;
  imageUrl: string;
}) {
  return (
    <div className="bg-white rounded-lg p-4 shadow-md">
      <h2 className="text-lg font-bold">{title}</h2>
      <p className="text-sm text-gray-600">{description}</p>
      <img src={imageUrl} alt={title} className="w-full h-40 object-cover" />
    </div>
  );
}
