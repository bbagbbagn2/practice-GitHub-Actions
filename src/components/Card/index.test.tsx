// src/components/Card/index.test.tsx

import { render } from "@testing-library/react";
import Card from "./index";

test("Card 컴포넌트가 정보를 올바르게 표시합니다", () => {
  const cardProps = {
    title: "제목",
    description: "설명",
    imageUrl: "https://example.com/image.jpg",
  };

  const { container } = render(<Card {...cardProps} />);
  expect(container).toMatchSnapshot();
});
