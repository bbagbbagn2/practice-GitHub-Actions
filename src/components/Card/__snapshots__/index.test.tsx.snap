// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Card 컴포넌트가 정보를 올바르게 표시합니다 1`] = `
<div>
  <div>
    <h2>
      제목
    </h2>
    <p>
      설명
    </p>
    <img
      alt="제목"
      src="https://example.com/image.jpg"
    />
  </div>
</div>
`;
